### DISCUSSION TITLE
Power (^) calculation is incorrect

### DISCUSSION THREAD

**Discussion by danemedina** [arendst/Tasmota#20964] [Q&A] [answered]
### PROBLEM DESCRIPTION
After compiling the base tasmota.bin build with #define USE_EXPRESSION, the power expression incorrectly calculates. var1=2^2 produces   "3.884" rather than the expected value of "4.0"

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _Generic (18)
- [x] Tasmota binary firmware version number used: _13.4.0.2 (tasmota)
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: _Firmware upgrade via the Tasmota web UI
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
14:51:55.482 MQT: stat/light/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"416":"PWM1"},"GPIO5":{"419":"PWM4"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"417":"PWM2"},"GPIO13":{"420":"PWM5"},"GPIO14":{"418":"PWM3"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
14:53:19.959 MQT: stat/light/STATUS11 = {"StatusSTS":{"Time":"2024-03-15T14:53:19","Uptime":"0T00:15:43","UptimeSec":943,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Dimmer":24,"Color":"3D15000000","HSBColor":"20,100,24","White":0,"CT":153,"Channel":[24,8,0,0,0],"Scheme":0,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"Hanson","BSSId":"F8:79:0A:43:D2:A3","Channel":1,"Mode":"11n","RSSI":74,"Signal":-63,"LinkCount":1,"Downtime":"0T00:00:05"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
14:54:51.424 CMD: var1=2^2
14:54:51.427 SRC: WebConsole from 10.0.0.167
14:54:51.429 CMD: Grp 0, Cmd 'VAR', Idx 1, Len 4, Pld -99, Data '=2^2'
14:54:51.436 MQT: stat/light/RESULT = {"Var1":"3.884"}


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
From console enter: var1=2^2

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
Resule for Var1 equal to 4.0

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
Tried various power expression and all failed to provide the correct result

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
To avoid including sizable math libraries, Tasmota is using an approximation for raising a number to a power, and while that approximation is "good enough" for some use cases, it is not generally usable if expecting accurate results. Better use other methods if feasible at all, like var1*var1 if already having the value in var1.
